Add inventory advanced section

joshuatf opened the pull request:
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Adds the advanced section for product inventory management.  Note that some styling changes around radios and radio labels exist in https://github.com/woocommerce/woocommerce/pull/35047.
<img width="464" alt="Screen Shot 2022-10-18 at 3 46 51 PM" src="https://user-images.githubusercontent.com/10561050/196559686-505ce94d-68b3-4e17-89c8-aeb360907960.png">



Closes #34389  .

### How to test the changes in this Pull Request:

1. Enable/disable the stock management for products at `wp-admin/admin.php?page=wc-settings&tab=products&section=inventory`
2. Under the product page (Products -> Add new (MVP)), click "Advanced" in the inventory section
3. Note that when stock management is enabled, the "out of stock" options exist
4. Note that in both cases, the limit 1 per purchase checkbox exists

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> run changelog add`?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

joshuatf committed:
Add sold individually inventory option

joshuatf committed:
Add backorder purchase options

joshuatf committed:
Add margin around collapsible content areas

joshuatf committed:
Add changelog entry

louwie17 reviewed (dismissed):
This tested well, and changes are simple and straightforward, nice work!
It would be great to add a couple tests around this though.

Approving this, there are some merge conflicts, so let me know when you need a re-approval.

joshuatf committed:
Add tests around inventory section

joshuatf committed:
Fix up checkbox props after rebase

joshuatf commented:
Thanks, @louwie17!  Rebased and also added tests in https://github.com/woocommerce/woocommerce/pull/35164/commits/e121f1a8d6fb1700b0500dad8487ed3ed911cbf9

joshuatf committed:
Check for disabled track quantity toggle

louwie17 commented on plugins/woocommerce-admin/client/products/sections/product-inventory-section/test/product-inventory-section.spec.tsx:
```diff
@@ -0,0 +1,175 @@
+		).toHaveClass( 'is-disabled' );
+	} );
+
+	it( 'should not render the manage stock section if inventory management is turned on', () => {
```
Should this actually be that the manage stock is not disabled?

joshuatf replied:
It should!  Thanks for the catch!

louwie17 reviewed (commented):
This is looking good @joshuatf, also appreciate you adding tests.
I did leave one comment in one of the tests, the name of it kinda confused me, in relation to what the test was actually doing.
Also when the track quantity is enabled it seems the margin is a lot bigger between the `Advanced ^` button and the `When out of stock` label, compared to the margin between `Restrictions` and the toggle when track quantity is disabled.
See screenshot:
<img width="713" alt="Screen Shot 2022-10-27 at 2 52 45 PM" src="https://user-images.githubusercontent.com/2240960/198363455-44f0b65b-9e99-4692-80b5-902636ea92ee.png">

joshuatf committed:
Update manage stock test

joshuatf committed:
Fix nested radio control label margin

joshuatf commented:
Thanks for the review, @louwie17.  I must have screwed up the rebase and created the margin issue.  Should be resolved now in https://github.com/woocommerce/woocommerce/pull/35164/commits/5db6f0707f0f7bcc6efba546c9b2afe12d6ddf47.

louwie17 reviewed (dismissed):
Styling looks good now, nice work @joshuatf , this looks good!

Btw you will likely have to rebase again with latest of trunk to fix the code sniff error.

louwie17 reviewed (approved):
LGTM 🚀
